File: snake-scala-2d/src/main/scala/metacake/scala/snake/entity/Food.scala
package metacake.scala.snake.entity

import java.awt.{Color, Graphics2D}

import io.metacake.core.output.RenderingInstruction
import io.metacake.s2d.output.drawing.instructions.{DrawInstruction, PlaceInstruction, RectangleInstruction}

object Food {
  val COLOR: Color = Color.RED
  val SIZE: Int = 50
  val IMAGE: DrawInstruction = new RectangleInstruction(Food.SIZE, Food.SIZE, Food.COLOR)
}

class Food(val x: Int, val y: Int) extends Positionable with Drawable {
  override def draw(scene: DrawInstruction): RenderingInstruction[Graphics2D] = {
    new PlaceInstruction(Food.IMAGE, x * Food.SIZE, y * Food.SIZE, scene)
  }
}

File: java-snake-2d/src/main/java/metacake/snake/entity/Food.java
package metacake.snake.entity;

import io.metacake.s2d.output.drawing.instructions.DrawInstruction;
import io.metacake.s2d.output.drawing.instructions.PlaceInstruction;
import io.metacake.s2d.output.drawing.instructions.RectangleInstruction;

import java.awt.*;

public class Food extends Positionable implements Drawable {

    private static final Color COLOR = Color.RED;
    public static final int SIZE = 50;
    private static final DrawInstruction IMAGE = new RectangleInstruction(SIZE, SIZE, COLOR);

    public Food(int x, int y) {
        super(x, y);
    }

    @Override
    public DrawInstruction draw(DrawInstruction instruction) {
        return new PlaceInstruction(IMAGE, getX() * SIZE, getY() * SIZE, instruction);
    }
}
